packages/cozy-client/src/models/contacts.spec.js

```diff
@@ -0,0 +1,87 @@
+import { createFullname, createSort, createDisplayName } from './contacts'
+
+const testContacts = [
+  {
+    name: { givenName: 'John', familyName: 'Smith' },
+    fullname: 'John Smith',
+    email: [{ address: 'john.smith@cozy.cc' }, { address: 'john.two@cozy.cc' }],
+    cozy: [{ url: 'https//www.john.com' }, { url: 'https//www.smith.com' }]
+  },
+  {
+    name: { givenName: '', familyName: 'Smith' },
+    fullname: 'Smith',
+    email: [{ address: '' }],
+    cozy: [{ url: '' }]
+  },
+  {
+    name: { givenName: 'John', familyName: '' },
+    fullname: 'John',
+    email: [{ address: '' }],
+    cozy: [{ url: '' }]
+  },
+  {
+    name: { givenName: '', familyName: '' },
+    fullname: '',
+    email: [],
+    cozy: [{ url: 'https//www.smith.com' }]
+  },
+  {
+    name: {},
+    fullname: '',
+    email: [],
+    cozy: []
+  },
+  {
+    name: { givenName: 'John' },
+    fullname: 'John',
+    cozy: [{ url: 'https//www.smith.com' }]
+  },
+  {
+    name: {},
+    email: [{ address: 'john.smith@cozy.cc' }],
+    cozy: [{ url: '' }]
+  }
+]
+
+describe('Contacts Model', () => {
+  it('should create correct fullname', () => {
+    const expected = ['John Smith', 'Smith', 'John', '', '', 'John', '']
```

Comment:
Nit: having the expectations next to the contact objects would make it easier to verify and maintain them :)

Reply:
En fait au lieu de faire X tests par fonction, j'ai fait un seul test qui boucle sur X cas. De plus la référence des tests et la même pour chaque fonction testée... mais ce n'est peut-être pas le plus lisible.

Je suis revenu à X tests par fonction, et chaque test embarque sa propre référence à tester.